src/main/java/com/pingcap/tikv/region/RegionManager.java

```diff
@@ -192,10 +201,12 @@ public void onRequestFail(long regionID, long storeID) {
               invalidateRegion(regionID);
             }
           } catch (InterruptedException | ExecutionException e) {
-            e.printStackTrace();
+            invalidateRegion(regionID);
           }
           // store's meta may be out of date.
           invalidateStore(storeID);
         });
+    // missing one last step here, need remove these region cache's store id is store id
+    // in go code, remove region cache share same store id.
```

Comment:
shall we just remove all cache? or it does not matter since load region is lazy.

Reply:
lazy is for first time and to perform as async. revisit will be cached.

Reply:
Yeah. that is my concern too. I could not find a way to invalidate element associated with same store id from `LoadCache`.

Reply:
according to loadingcache's semantics, we might change it to eager loading, and if exception happens, just don't change cache status.

Reply:
LoadCache doest not fit the semantics in this particular case.  We may need write our own cache. That is rather quite lot work. I guess the temporary solution we have is invalidate all cache. At lease, this way will not give us any error.